src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Configuration/WattTimeClientConfigurationTests.cs

```diff
@@ -8,6 +8,7 @@ namespace CarbonAware.DataSources.WattTime.Tests;
 public class WattTimeClientConfigurationTests
 {
     [TestCase("testuser", "12345", "http://example.com", TestName = "Validate does not throw: username; password; url")]
+    [TestCase("testuser", "pass£word", "http://example.com", TestName = "Validate throws: username; non-ASCII password; url")]
```

Comment:
Is it possible to add a test to inject non-utf8? So the `ConfigurationException` path can be tested.

Reply:
I don't know, I've been searching but can't find a string. Seems like it'd have to write a separate test with Unicode bytes to try to see the failure, doesn't seem worth it imho

Reply:
> I don't know, I've been searching but can't find a string. Seems like it'd have to write a separate test with Unicode bytes to try to see the failure, doesn't seem worth it imho

In the test case `Validate_Throws` you can add the following test cases:
```c#
    [TestCase("\ud800", null, "http://example.com", TestName = "Non utf8 characters for username")]
    [TestCase(null, "\ud83e\udd70", "http://example.com", TestName = "Non utf8 characters for password")]
```